### ISSUE TITLE
Deletion of a template associated with a at least one group should not be possible

### ISSUE THREAD

**Issue by kwilczynski** [joyent/triton-service-groups#68]
At the moment, we allow for a template to be deleted, even though it might still be associated with a group. This results in an error during a group removal as per:

```
HTTP/2 500
content-type: text/plain; charset=utf-8
date: Sun, 15 Apr 2018 21:17:32 GMT
strict-transport-security: max-age=31536000; includeSubdomains; preload
x-content-type-options: nosniff
content-length: 29

Error finding template by ID
```
And in an orphaned job at the Nomad side which is still maintaining the desired compute instance count for the length of the lifespan of such job, making it impossible for the end-user to terminate instance using the API et al.

**Steps to reproduce:**

1) Create a valid template
2) Create a group (with a capacity of 1) using the template unique identifier (ID) from step (1)
3) Allow for an instance created by step (2) to be provisioned and running
4) Delete the template created in the step (1)

Following step (4), you will be able to observe a compute instance still being present and maintained (it will be replaced with new instance should the old one be killed, etc.). It will be almost impossible to remove such instance (or a number of them) for a long as the Nomad job runs.

**Comment by kwilczynski** [comment]
A possible solution would include checking if a template is used by at least a single existing and current group, and that it hasn't been archived - if that is the case, we should issue a `409 Conflict` and a message saying that this particular template is used by a group (or a list of groups using it).

Templates are immutable and cannot be updated (one has to create a new template with different parameters, etc.), therefore making it impossible to remove a template being used by a number of groups (or at least a single group) is not unreasonable, given that it cannot be manipulated with anyway.

**Comment by kwilczynski** [comment]
To add:

It is also possible to remove a group with an arbitrary template ID, regardless of whether a template existed prior or not. This results in the same `500` HTTP error as outlined above, and the groups is also removed, leaving orphaned job which is still maintaining resources.

Additionally, it is also possible to add a group with a non-existent template. It does result in an error being sent back (which is `500` HTTP error), but a group is being added nonetheless. This results in a no new compute instances being created, which is expected.

**Comment by kwilczynski** [comment]
The Pull Request #76 which was merged made it impossible to add a new group without a non-existent template. Which is also a huge improvement for this issue.

**Comment by stack72** [comment]
Closed via #88
